functions/_common/analytics.ts

```diff
@@ -0,0 +1,133 @@
+import { parseCookies } from "./cookieTools.js"
+import { Env, extensions } from "./env.js"
+import { uuidv7 } from "uuidv7"
+
+export async function sendEventToGA4(
+    event: { name: string; params: Record<string, string | number> },
+    request: Request,
+    env: Env
+) {
+    if (!validateAnalyticsEnvVariables(env)) {
+        return
+    }
+    const cookies = parseCookies(request)
+    const client_id =
```

Comment:
I'm not sure this will work correctly, yet. Need to test it with real GTM cookies.